service/service.go

```diff
@@ -442,7 +442,16 @@ func (s *service) UpdateProductCategory(ctx context.Context, prodCategory *prodE
 	return s.productRepository.UpsertProductCategory(ctx, prodCategory)
 }
 
-func (s *service) UpsertProductType(ctx context.Context, prodType *prodEntity.ProductType) error {
+func (s *service) UpsertProductType(ctx context.Context, prodType *prodEntity.ProductType, isUpdate bool) error {

```

Comment:
why is the parameter named with isUpdate? could we change to another name that is related to the condition?

Reply:
in grpc handler, both insert and update functions called this UpsertProductType function in the service package, so isUpdate parameter is used to differentiate whether the request is to update the product type (isUpdate = true) or to insert product type (isUpdate = false).
There will be an additional validation if the request is an update to the product type.
Do you have any suggestions for the variable name mas?

Reply:
can we check from the ID? I mean when we are inserting data, the ID is not provided and when updating data, the ID is provided from the FE. WDYT mas?

Reply:
Yes I agreed and we can actually do that also, okay will try to apply that one, thank you mas

Reply:
Updated ya mas